### PR TITLE
Fix switch-ethereum-chain handler by passing configuration id to setA…

### LINKED ISSUE #18453: [Bug]: RPC Error when switching network on 10.28.1
### Describe the bug

I just updated to 10.28.1 and I get an error when trying to switch networks from various dapps, including app.aave.com. I am able to add the network, but if I have previously added the network and just try to switch to the network from the dapp the console shows the following message: `RPC Error: networkConfigurationId [object Object] does not match a configured networkConfiguration `

### Steps to reproduce

1. Install 10.28.1
2. Go to app.aave.com testnet
3. Go to a modal for a network that you aren't connected to (e.g. Fantom) and click on switch network. Add network by clicking "switch network".
4. Switch back to another network (e.g. Ethereum mainnet) and repeat step 3. This time, aave will show a cannot switch network error, and you will see aforementioned error in the developer console.

### Error messages or log output

```shell
RPC Error: networkConfigurationId [object Object] does not match a configured networkConfiguration
```

### Version

10.28.1

### Build type

None

### Browser

Chrome

### Operating system

MacOS

### Hardware wallet

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by danjm**
Fixes https://github.com/MetaMask/metamask-extension/issues/18453

## Explanation

In a previous commit, we updated `setActiveNetwork` in the network controller to receive a network configuration id instead of a whole network object. However, the parameter passed to `setActiveNetwork` in the `switch-ethereum-chain.js` handler was still passed the object, not the id. See https://github.com/MetaMask/metamask-extension/commit/23ca4460cf#diff-93f17dab5026baf840aec766ac5b6ebcd35d5bb0abb45e2c3f41791423f2a488R117

This PR corrects that, and adds an e2e test for the `wallet_switchEthereumChain` functionality, which would have caught this bug.

It is worth noting that the bug this PR fixes would not occur if you switched to the network immediately after adding it, because the correct parameter is passed to the `setActiveNetwork` call in `add-ethereum-chain.js` which happens after prompting the user to switch networks (https://github.com/MetaMask/metamask-extension/commit/23ca4460cf#diff-b3485a1ea9943fd380364f73a53295289f2ccd3817423d4f245b5102f8620b3cR170)

## Manual Testing Steps

1. Run `$(yarn bin ganache) --port 8546 --chain.chainId 1338`
2. Go to https://metamask.github.io/test-dapp/ and connect to the dapp
3. Click "Add Localhost 8546" and "Approve", BUT click "Cancel" once you get to the "Allow this site to switch the network?" screen (so that if you were to open MetaMask, the Localhost 8546 network would not be selected)
4. Click "Switch to Localhost 8546" and then "Approve"
5. If you open MetaMask, the Localhost 8546 network should now be selected

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by pedronfigueiredo — DISMISSED** [review]

File: test/e2e/tests/switch-custom-network.spec.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        await driver.switchToWindow(extension);
+
+        const currentNetworkName = await driver.findElement({
+          tag: 'span',
+          text: 'Localhost 8546',
+        });
+
+        assert.ok(
+          Boolean(currentNetworkName),
+          'Failed to switch to custom network',
```
Comment (DDDDDanica) [inline test/e2e/tests/switch-custom-network.spec.js:97]: I'm a bit confused about. the assertion here, we should switch to localhost network successfully according to the test title and the actions

File: app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js [hunk 2]
```diff
@@ -115,7 +115,7 @@ async function switchEthereumChainHandler(
       ) {
         setProviderType(approvedRequestData.type);
       } else {
-        await setActiveNetwork(approvedRequestData);
+        await setActiveNetwork(approvedRequestData.id);
```
[the remark this answers is not in the document]
Comment (danjm) [reply app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js:118]: yeah, codecov can't measure e2e tests

File: test/e2e/tests/switch-custom-network.spec.js (hunk 1 above)
Comment (DDDDDanica) [reply test/e2e/tests/switch-custom-network.spec.js:97]: I understand. but here we are expecting a success switch.? or am i missing something ..

Comment (danjm) [reply test/e2e/tests/switch-custom-network.spec.js:97]: if `currentNetworkName` is defined, it means that we switched successfully to the "Localhost 8546" network. If it is undefined, it means that we did not. So this assertion verifies that we successfully switched to the network

**Review by DDDDDanica — DISMISSED** [review]

**Review by DDDDDanica — DISMISSED** [review]

**Comment by danjm** [comment]
reverted the last commit because it broke lint (no arrow functions in `describe`)

**Review by DDDDDanica — APPROVED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**